resources/bundles/org.eclipse.core.resources/src/org/eclipse/core/internal/localstore/FileSystemResourceManager.java

```diff
@@ -1196,6 +1193,13 @@ public void write(IFile target, InputStream content, IFileInfo fileInfo, int upd
 		}
 	}
 
+	private void assertExists(IFile target) {
+		if (!target.exists()) {
```

Comment:
If I understand correctly, this will only check the workspace tree for the existence, not the file system itself. It only evaluated the resource info of the file. Access to the resource info is performed several times during the `write` method, so there should not be a noticable overhead, at least when assuming the `exists` implementation of `Resource`. However, we may reduce this to only check the resource info for existence like the existing code does. Then we can reuse the already existing access to the resource info and only need to add one or two further accesses, which will not produce a relevant overhead.